src/components/Item/TodoItem.tsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable max-len */
+import React, { useState, useRef } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { updateTodo } from '../../api/todos';
+import { handleError } from '../../handleError';
+import { ErrorMessage } from '../../types/ErrorMessage';
+
+type TodoItemProps = {
+  todo: Todo;
+  handleDelete: (todoId: number) => void;
+  onStatusChange: (todoId: number, completed: boolean) => void;
+  onTodoUpdate: () => void;
+};
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo, handleDelete, onStatusChange, onTodoUpdate,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const [errorMessage, setErrorMessage] = useState('');
+  const editRef = useRef<HTMLInputElement>(null);
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setIsEditing(false);
+    onStatusChange(todo.id, todo.completed);
+    if (newTitle.trim() === '') {
+      handleDelete(todo.id);
+    } else if (newTitle !== todo.title) {
+      try {
+        await updateTodo(todo.id, {
+          title: newTitle.trim(),
+        });
+        onStatusChange(todo.id, todo.completed);
+        onTodoUpdate();
+      } catch {
+        handleError(setErrorMessage, ErrorMessage.noUpdateTodo);
+      }
+    }
+  };
+
+  const handleBlur = () => {
+    setIsEditing(false);
+  };
```

Comment:
On blur we should:
- Do nothing only when title has not changed.
- Update title when it has changed
- Remove Todo when title left empty
This logic is missing here